tests/resources/test_gemd_resource.py

```diff
@@ -57,11 +57,29 @@ def session() -> FakeSession:
 @pytest.fixture
 def gemd_collection(session) -> GEMDResourceCollection:
     return GEMDResourceCollection(
-        project_id=uuid4(),
+        team_id=uuid4(),
         dataset_id=uuid4(),
         session=session
     )
 
+@pytest.fixture
+def invalid_collection(session) -> None:
+    with pytest.raises(TypeError):
+        return GEMDResourceCollection(
+            dataset_id=UUID('8da51e93-8b55-4dd3-8489-af8f65d4ad9a'),
+            session=session,
+            )
+
+def test_invalid_collection_construction(invalid_collection):
+    # assertion is within the construction of the invalid_collection
```

Comment:
Fixtures should be used to construct data. If the construction is the test, it belongs in the test function, unless there's some very good reason.

Reply:
Got it. I'll fix this and my other occurrences of a similar pattern